bot/exts/utils/jams.py

```diff
@@ -20,124 +22,153 @@ class CodeJams(commands.Cog):
+    async def codejam(self, ctx: commands.Context) -> None:
+        """A Group of commands for managing Code Jams."""
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @codejam.command()
+    async def create(self, ctx: commands.Context, csv_file: t.Optional[str]) -> None:
         """
-        Create team channels (voice and text) in the Code Jams category, assign roles, and add overwrites for the team.
+        Create code-jam teams from a CSV file or a link to one, specifying the team names, leaders and members.
+
+        The CSV file must have 3 columns: 'Team Name', 'Team Member Discord ID', and 'Team Leader'.
 
-        The first user passed will always be the team leader.
+        This will create the text channels for the teams, and give the team leaders their roles.
         """
-        # Ignore duplicate members
-        members = list(unique_everseen(members))
-
-        # We had a little issue during Code Jam 4 here, the greedy converter did it's job
-        # and ignored anything which wasn't a valid argument which left us with teams of
-        # two members or at some times even 1 member. This fixes that by checking that there
-        # are always 3 members in the members list.
-        if len(members) < 3:
-            await ctx.send(
-                ":no_entry_sign: One of your arguments was invalid\n"
-                f"There must be a minimum of 3 valid members in your team. Found: {len(members)}"
-                " members"
-            )
-            return
+        async with ctx.typing():
+            if csv_file:
+                async with self.bot.http_session.get(csv_file) as response:
+                    if response.status != 200:
+                        await ctx.send(f"Got a bad response from the URL: {response.status}")
+                        return
 
-        team_channel = await self.create_channels(ctx.guild, team_name, members)
-        await self.add_roles(ctx.guild, members)
+                    csv_file = await response.text()
 
-        await ctx.send(
-            f":ok_hand: Team created: {team_channel}\n"
-            f"**Team Leader:** {members[0].mention}\n"
-            f"**Team Members:** {' '.join(member.mention for member in members[1:])}"
-        )
+            elif ctx.message.attachments:
+                csv_file = (await ctx.message.attachments[0].read()).decode("utf8")
+            else:
+                raise commands.BadArgument("You must include either a CSV file or a link to one.")
+
+            teams = defaultdict(list)
+            reader = csv.DictReader(csv_file.splitlines())
+
+            for row in reader:
+                member = ctx.guild.get_member(int(row["Team Member Discord ID"]))
+
+                if member is None:
+                    log.trace(f"Got an invalid member ID: {row['Team Member Discord ID']}")
```

Comment:
I think a higher log level would be useful here, or maybe even send a message to the channel saying that it contains the following invalid members and then it lists down all of them.

Reply:
I think it would be good if we keep track of members who could not be found in a list, and then send them in an embed at the end. This will allow the person running them to easily know who needs to be manually contacted.

Reply:
But also, the fact that J9 said below that she's operating on the assumption that this csv is valid for now, I don't think we need this for this PR.

Reply:
Alright sounds good 👍🏾